libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Functions.hs

```diff
@@ -330,11 +373,37 @@ alwaysFalse p@(Allegra _) _ n = never n p
 alwaysFalse p@(Shelley _) _ n = never n p
 
 certs :: Proof era -> Core.Tx era -> [DCert (Crypto era)]
-certs (Babbage _) tx = toList $ getField @"certs" (getField @"body" tx)
-certs (Alonzo _) tx = toList $ getField @"certs" (getField @"body" tx)
-certs (Mary _) tx = toList $ getField @"certs" (getField @"body" tx)
-certs (Allegra _) tx = toList $ getField @"certs" (getField @"body" tx)
-certs (Shelley _) tx = toList $ getField @"certs" (getField @"body" tx)
+certs (Babbage _) tx = Fold.toList $ getField @"certs" (getField @"body" tx)
+certs (Alonzo _) tx = Fold.toList $ getField @"certs" (getField @"body" tx)
+certs (Mary _) tx = Fold.toList $ getField @"certs" (getField @"body" tx)
+certs (Allegra _) tx = Fold.toList $ getField @"certs" (getField @"body" tx)
+certs (Shelley _) tx = Fold.toList $ getField @"certs" (getField @"body" tx)
+
+createRUpdNonPulsing' ::
+  forall era.
+  Proof era ->
+  Model era ->
+  RewardUpdateOld (Crypto era)
+createRUpdNonPulsing' proof model =
+  let bm = BlocksMade $ mBcur model -- TODO or should this be mBprev?
+      ss = mSnapshots model
+      as = mAccountState model
+      reserves = _reserves as
+      pp = mPParams model
+      totalStake = Map.foldr (<+>) (Coin 0) $ mRewards model
```

Comment:
That's just a `fold`. Just an observation, no need to change though.
```suggestion
      totalStake = fold $ mRewards model
```

Reply:
ok